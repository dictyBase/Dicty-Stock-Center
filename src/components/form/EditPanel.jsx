import React, { Component, PropTypes } from 'react'
import 'styles/core.scss'

export default class EditPanel extends Component {
    displayName = 'panel to edit order info';

    static propTypes = {
        user: PropTypes.object.isRequired,
        edit: PropTypes.func.isRequired,
        title: PropTypes.string
    }
    render() {
        const { title, edit } = this.props
        const { firstName, lastName, address, address2,
            city, state, zip, country
        } = this.props.user
        return (
            <div>
                <h4>{ title }</h4>
                <div>{ firstName } { lastName }</div>
                <div>{ address }</div>
                <div>{ address2 && address2 }</div>
                <div>{ city } { state && state } { zip }</div>
                <div>{ country }</div>
                <a href="#" onClick = { edit }>Edit</a>
            </div>
        )
    }
}
