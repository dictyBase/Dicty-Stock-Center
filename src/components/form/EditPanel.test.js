import React from 'react'
import renderer from 'react-test-renderer'
import EditPanel from './EditPanel'

test('matching a snapshot of EditPanel', () => {
    const user = {
        firstName: 'Jane',
        lastName: 'Doe'
    }
    const address = {value: 'n michigan', touched: true}
    const address2 = {value: 'floor 1', touched: true}
    const city = {value: 'chicago', touched: true}
    const state = {value: 'IL', touched: true}
    const zip = {value: '65222', touched: true}
    const country = {value: 'USA', touched: true}
    const edit = () => {}

    const component = renderer.create(
    <EditPanel
        user={ user }
        firstName={ user.firstName }
        lastName={ user.lastName }
        address={ address }
        address2={ address2 }
        city={ city }
        state={ state }
        zip={ zip }
        country={ country }
        edit={ edit }
    />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
