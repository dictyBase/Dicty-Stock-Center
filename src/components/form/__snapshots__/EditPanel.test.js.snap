// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of EditPanel 1`] = `
<div>
  <h4 />
  <div>
    Jane
     
    Doe
  </div>
  <div />
  <div />
  <div>
     
     
  </div>
  <div />
  <br />
  <div>
    <a
      className="sc-kUaPvJ hsqoua"
      onClick={[Function]}
    >
      <span
        aria-hidden={true}
        className="fa fa-pencil-square-o"
      />
       Edit shipping info
    </a>
  </div>
</div>
`;
